### ISSUE TITLE
Always show file extensions

### ISSUE THREAD

**Issue by jeffgca** [fission-suite/drive#128]
NB: Feature requests will only be considered if they solve a pain

# Summary

Most OS file explorers show file extensions. So should we.

## Problem

* Files in drive don't display file extensions, but the resulting urls do.
* I can't create a .yml file in drive without having an extra fle extension added to it.
* I can't create files without file extensions.

### Impact

Some confusion between drive and what users would see on disk wrt what file extensions are used.

## Solution

Don't automagically add file extensions when creating files in drive itself, just let the user supply the file name including file extension.

When a user creates a new file in drive, assume it is a text file with encoding utf8 and have the user add whatever file extension they want.

**Comment by icidasset** [comment]
File extensions are shown when hovering over the directory entry, and on the detail view (sidebar when file is selected). I don't want to show all extensions by default, but I do want to make this an option later.

I agree with the extension handling when creating files 👍 

I suggest the following:
- Add a "new file" button, doesn't change filename at all
- Hide "suggested extension" buttons when the filename already has an extension

I don't really like how these buttons fit into the whole sidebar currently, might change this up in the future.

**Comment by therealjeffg** [comment]
> I don't want to show all extensions by default, but I do want to make this an option later.

Can you unpack for me what the reasoning is here? Generally my ethic with experiences that closely follow established UI conventions is to follow all of the conventions as much as possible so as to prevent confusion.

**Comment by icidasset** [comment]
Definitely. I'm following the MacOS convention. Don't show the file extension by default, and rely on the icon and the name to identify the file. Two differences though: (1) I think MacOS shows the file extension for unrecognised file extensions (ie. no icon) (2) We show the extension on hover.

In addition to that, I think it looks nicer, and feel that the icon + name is enough to identify the file.
But like I said, I do want to add the option later.

**Comment by icidasset** [comment]
@therealjeffg Thoughts on the current file/dir creation? Does it solve all the problems you had with it?

**Comment by therealjeffg** [comment]
> @therealjeffg Thoughts on the current file/dir creation? Does it solve all the problems you had with it?

Sorry, I neglected to get back to this issue.

My opinion, perhaps put more simply: while it *may be* confusing to some people if we don't show the full name of the file for known file types, it will never be confusing to them if we do show the full name of the file. 

_Disclaimer: I am biased here, I am used to looking at file systems in a terminal and always enable "show file extensions" preferences on any computer I use._

**Comment by icidasset** [comment]
Sounds reasonable 👍  I'll make it a setting.

**Comment by therealjeffg** [comment]
> Sounds reasonable 👍 I'll make it a setting.

Questions: 

1. does drive have other settings? I don't think this should be the only setting
2. for the user, the least surprising thing to do here is to always show file extensions by default.
